### PR TITLE
Analysis methods

### PR DESCRIPTION
**Pull request by NotoriousBIF**
Hi Niek, zou je misschien feedback willen geven op de functie generate_results? Ik combineer de functies die je me eerder toe hebt gestuurd.

### CODE REVIEW

File: orbitrap_training.py [hunk 1]
```diff
@@ -19,16 +19,15 @@
 orbitrap_regexp_terms = re.compile(".*orbitrap.*|.*hcd.*|.*q-exactive.*|.*lumos.*|.*velos.*", re.IGNORECASE)
```
Comment (niekdejonge) [inline orbitrap_training.py:19]: This script can be a bit cleaner/clearer. 
You can use more functions that can be generalized and therefore reused for the other models as well. So for instance a function for training a model. 

After creating these generalized functions, I think it is best to create a jupyter notebook for selecting and training. You can in that case make 1 notebook for all 4 specific models and still keep it clear, by adding clear markdown. Using 4 scripts that have the code behind if __name__ == "__main__":  is also a fine alternative.

File: orbitrap_training.py [hunk 2]
```diff
@@ -66,7 +65,7 @@
                           epochs=150, verbose=1, callbacks=[earlystopper_scoring_net])
 
 model_path = 'G:/Remco Bsc Thesis/Models/'
-model_file_name = os.path.join(model_path, "ms2deepscore_model_orbitrap.hdf5")
+model_file_name = os.path.join(model_path, "ms2deepscore_model_orbitrap_v2.hdf5")
 model.save(model_file_name)
 
 from matplotlib import pyplot as plt
```
Comment (niekdejonge) [inline orbitrap_training.py:71]: Make a general function for plotting and import statements at the top

File: analysis_methods.py [hunk 3]
```diff
@@ -3,45 +3,25 @@
...
-
 def create_sets(master_subset: list, size_training: float, size_validation: float, size_testing: float):
     """this method randomly splits a subset into user-defined portions for training, validation and testing.
     master_subset: Subset which you want to split into separate parts
     size_training: Size of desired training set. Expected argument is percentage as float e.g. 0.8
     size_validation: Size of desired validation set. Same expectancies as size_training
     size_testing: Size of desired testing set. Same expectancies as size_training
-    Returns: a dictionary with each subset sorted under its own key"""
+    Returns: three lists of spectra, training_final, validation_final and testing_final"""
     testing_amount = math.floor(size_testing*len(master_subset))
```
[the remark this answers is not in the document]
Comment (NotoriousBIF) [reply analysis_methods.py:19]: Thanks for the suggestion! I added it :)

File: analysis_methods.py [hunk 4]
```diff
@@ -123,7 +104,90 @@ def calculate_errors_pairs(list_of_scores):
...
+        output_list.append(current_score)
+    return output_list
+
+from typing import List
+from matchms import Spectrum
+import pandas as pd
+import numpy as np
+
+def select_predictions_for_test_spectra(tanimoto_df: pd.DataFrame,
+                                        test_spectra: List[Spectrum]) -> np.ndarray:
```
[the remark this answers is not in the document]
Comment (NotoriousBIF) [reply analysis_methods.py:33]: Will start using this in the future

File: orbitrap_training.py (hunk 1 above)
Comment (NotoriousBIF) [reply orbitrap_training.py:19]: Yes! I initially wanted to make a more general function for training but decided to go with these scripts first. I will try making a more general function later on

File: analysis_methods.py [hunk 5]
```diff
@@ -123,7 +104,90 @@ def calculate_errors_pairs(list_of_scores):
...
+     tanimoto_df: the dataframa containing the true scores
+     amount_of_bins: the amount of bins you want to distribute the errors over
+     returns: list with the output from tanimoto_independent_losses and the average RMSE"""
+    bins = calculate_bins(amount_of_bins)
+    similarity_measure = MS2DeepScore(ms2deepscore_model)
+    predicted_scores = similarity_measure.matrix(testspectra[:], testspectra[:])
+    true_scores = select_predictions_for_test_spectra(tanimoto_df, testspectra)
+    bin_content, bounds, rmses, maes = tanimoto_dependent_losses(predicted_scores, true_scores, bins)
+    global_rmse = sum(rmses)/len(rmses)
+    output = [bin_content, bounds, rmses, maes, global_rmse]
```
[the remark this answers is not in the document]
Comment (NotoriousBIF) [reply analysis_methods.py:99]: I am not quite sure whether I will still use these for a plot, so for the moment I will leave them in

**Pull request closed without merging**